### PR TITLE
detectBadPaginationMiddleware

### PR DESCRIPTION
**Pull request by davecoates**

### CODE REVIEW

File: doc-site/package.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
   "version": "0.1.0",
   "private": true,
   "scripts": {
-    "dev": "next dev",
+    "dev": "next dev -p 4000",
```
Comment (fanglong-alliance) [inline doc-site/package.json:6]: whats this? o.O

Comment (davecoates) [reply doc-site/package.json:6]: run the doc site on a different port (defaults to 3000... same as webpack server)

File: js-packages/@prestojs/util/src/pagination/CursorPaginator.ts [hunk 2]
```diff
@@ -199,4 +200,36 @@ export default class CursorPaginator extends Paginator<
...
+        if (Array.isArray(decodedBody) || !decodedBody) {
+            return false;
+        }
+
+        // Cursor pagination responses should contain a next and previous link that includes
+        // a query parameter with name 'cursor' containing the next/previous cursor value. Results
+        // should be an array under the `results` key.
+        const isCursorLink = (link: any): boolean =>
+            typeof link == 'string' && 'cursor' in qs.parse(qs.extract(link));
+        if (isCursorLink(decodedBody.next) || isCursorLink(decodedBody.previous)) {
```
Comment (fanglong-alliance) [inline js-packages/@prestojs/util/src/pagination/CursorPaginator.ts]: what happens if both next & previous are null? (ie. paginated but single page)

shouldnt the middleware still treat it as paginated?

File: js-packages/@prestojs/util/src/pagination/PageNumberPaginator.ts [hunk 3]
```diff
@@ -50,6 +53,12 @@ export default class PageNumberPaginator extends Paginator<
         if (typeof page === 'string') {
             return Number(page);
         }
+        if (page == null) {
```
Comment (fanglong-alliance) [inline js-packages/@prestojs/util/src/pagination/PageNumberPaginator.ts:56]: should this be a === just because?

also, yay

Comment (davecoates) [reply js-packages/@prestojs/util/src/pagination/PageNumberPaginator.ts:56]: want to catch either undefined or null

File: js-packages/@prestojs/util/src/pagination/CursorPaginator.ts (hunk 2 above)
Comment (davecoates) [reply js-packages/@prestojs/util/src/pagination/CursorPaginator.ts]: yes good catch - existing code was buggy in this way

i've pushed a fix now

**Review by fanglong-alliance — APPROVED** [review]
lgtm

**Pull request merged**